windows/src/desktop/setup/Keyman.Setup.System.InstallInfo.pas

```diff
@@ -377,50 +365,36 @@ procedure TInstallInfo.LocatePackagesAndTierFromFilename(Filename: string);
   end;
 end;
 
-procedure TInstallInfo.LocatePackagesInPath(const path: string);
+procedure TInstallInfo.LoadLocalPackagesMetadata;
 var
-  f: TSearchRec;
-  id, p: string;
   pack: TInstallInfoPackage;
   packLocation: TInstallInfoPackageFileLocation;
   lang: TPackageKeyboardLanguage;
   iipl: TInstallInfoPackageLanguage;
 begin
-  p := IncludeTrailingPathDelimiter(path);
-  if FindFirst(p+'*'+Ext_PackageFile, 0, f) = 0 then
+  for pack in FPackages do
   begin
-    repeat
-      if SameText(ExtractFileExt(f.Name), Ext_PackageFile) then
+    for packLocation in pack.Locations do
+    begin
+      if (packLocation.LocationType = iilLocal) and GetPackageMetadata(packLocation.Path, packLocation.LocalPackage) then
       begin
-        packLocation := TInstallInfoPackageFileLocation.Create(iilLocal);
-        if GetPackageMetadata(p + f.Name, packLocation.LocalPackage) then
-        begin
-          id := ChangeFileExt(f.Name, '').ToLower;
+        packLocation.Size := FileSizeByName(packLocation.Path);
+        packLocation.Name := packLocation.LocalPackage.Info.Desc[PackageInfo_Name];
+        packLocation.Version := packLocation.LocalPackage.Info.Desc[PackageInfo_Version];
 
-          pack := FPackages.FindById(id, True);
-          packLocation.Size := FileSizeByName(p + f.Name);
-          packLocation.Name := packLocation.LocalPackage.Info.Desc[PackageInfo_Name];
-          packLocation.Version := packLocation.LocalPackage.Info.Desc[PackageInfo_Version];
-          packLocation.Path := p + f.Name;
-
-          if packLocation.LocalPackage.Keyboards.Count = 1 then
+        if packLocation.LocalPackage.Keyboards.Count = 1 then
```

Comment:
Do you also want to filter for .kmx keyboards so setup doesn't display keyboard packages that only contain touch layout keyboards?

Reply:
That's a good idea but I think low priority.